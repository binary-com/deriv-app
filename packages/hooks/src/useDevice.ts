import { useWindowSize } from 'usehooks-ts';

const BREAKPOINTS = {
    maxMobile: 600,
    maxTablet: 1280,
};

/** A custom hook to check for the client device and determine the layout to be rendered */
const useDevice = () => {
    const { width } = useWindowSize();
    const isMobile = width > 0 && width < BREAKPOINTS.maxMobile;
    const isTablet = width >= BREAKPOINTS.maxMobile && width < BREAKPOINTS.maxTablet;
    const isDesktop = width >= BREAKPOINTS.maxTablet;

    return {
        isDesktop,
        isMobile,
        isTablet,
    };
};

export default useDevice;
