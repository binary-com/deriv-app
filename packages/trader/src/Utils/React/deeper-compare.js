/**
 * Deep compare component's props and state for re-render check.
 *
 * @return {Boolean} Whether we should re-render the component or not
 */
export const deeperCompare = (next_props, next_state, this_props, this_state) => {
    let has_prop_changed = false;
    let has_state_changed = false;

    if (isComparable(next_props)) {
        has_prop_changed = Object.keys(next_props)
            .some(prop => typeof next_props[prop] !== 'function' &&
            typeof next_props[prop] === 'object' ?
                JSON.stringify(next_props[prop]) !== JSON.stringify(this_props[prop]) :
                next_props[prop] !== this_props[prop]);
    }

    if (isComparable(next_state)) {
        has_state_changed = Object.keys(next_state).some(state => next_state[state] !== this_state[state]);
    }

    return has_prop_changed || has_state_changed;
};

const isComparable = (value) => !!value &&  typeof value === 'object';
