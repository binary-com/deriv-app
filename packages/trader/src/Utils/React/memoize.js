export function memoize(func) {
    const cache = {};
    return function (...args) {
        const key = JSON.stringify(args);
        if (key in cache) return cache[key];
        return (cache[key] = func(...args));
    };
}
