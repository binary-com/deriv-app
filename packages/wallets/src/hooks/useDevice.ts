import { useWindowSize } from 'usehooks-ts';

/** A custom hook to check for the client device and determine the layout to be rendered */
const useDevice = () => {
    const { width } = useWindowSize();
    const isMobile = width > 0 && width <= 600;
    const isTablet = width > 600 && width < 1280;
    const isDesktop = width >= 1280;

    return {
        isDesktop,
        isMobile,
        isTablet,
    };
};

export default useDevice;
