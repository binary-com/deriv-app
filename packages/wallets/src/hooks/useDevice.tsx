import { useWindowSize } from 'usehooks-ts';

const useDevice = () => {
    const { width } = useWindowSize();
    const is_mobile = width < 768;
    const is_tablet = width >= 768 && width < 1024;
    const is_desktop = width >= 1024;

    return {
        is_mobile,
        is_tablet,
        is_desktop,
    };
};

export default useDevice;
