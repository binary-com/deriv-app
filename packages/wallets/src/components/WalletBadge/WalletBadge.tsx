import React from 'react';
import useDevice from '../../hooks/useDevice';
import { WalletText } from '../Base';
import './WalletBadge.scss';

type TWalletBadgeProps = {
    children: React.ReactNode;
};

const WalletBadge = ({ children }: TWalletBadgeProps) => {
    const { isMobile } = useDevice();
    return (
        <div className='wallets-badge'>
            <WalletText color='black' size={isMobile ? 'sm' : 'xs'}>
                {children}
            </WalletText>
        </div>
    );
};

export default WalletBadge;
