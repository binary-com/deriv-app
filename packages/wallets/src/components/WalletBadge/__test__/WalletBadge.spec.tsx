import React from 'react';
import { render, screen } from '@testing-library/react';
import useDevice from '../../../hooks/useDevice';
import WalletBadge from '../WalletBadge';

jest.mock('../../../hooks/useDevice');
const mockedUseDevice = useDevice as jest.MockedFunction<typeof useDevice>;

describe('WalletBadge Component', () => {
    beforeEach(() => {
        mockedUseDevice.mockReturnValue({ isDesktop: true, isMobile: false, isTablet: false });
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should render without crashing', () => {
        render(<WalletBadge>Test Child</WalletBadge>);
        expect(screen.getByText('Test Child')).toBeInTheDocument();
    });

    it(`should render correctly when isMobile is true`, () => {
        mockedUseDevice.mockReturnValue({ isDesktop: false, isMobile: true, isTablet: false });
        render(<WalletBadge>Test Child</WalletBadge>);
        const badge = screen.getByText('Test Child');
        expect(badge).toHaveClass('derivs-text__size--sm');
    });
});
