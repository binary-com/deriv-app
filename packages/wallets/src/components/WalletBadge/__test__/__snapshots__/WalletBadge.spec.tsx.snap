// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletBadge Component should render correctly when isMobile is true 1`] = `
<span
  class="wallets-text wallets-text__size--sm wallets-text__weight--normal wallets-text__align--left wallets-text__color--black wallets-text__line-height--undefined wallets-text__font-style--normal"
>
  Test Child
</span>
`;

exports[`WalletBadge Component should render correctly when isMobile is true} 1`] = `
<span
  class="wallets-text wallets-text__size--sm wallets-text__weight--normal wallets-text__align--left wallets-text__color--black wallets-text__line-height--undefined wallets-text__font-style--normal"
>
  Test Child
</span>
`;
